src/api/v4/error-handling/carto-error-extender.js

```diff
@@ -0,0 +1,82 @@
+var _ = require('underscore');
+var ERROR_LIST = require('./error-list');
+
+/**
+ * Get two parameters to enrich a CartoError.
+ * - friendlyMessage: A easy to understand error description.
+ * - errorCode: A unique errror code
+ * 
+ * @param {CartoError} cartoError
+ * 
+ * @returns {object} - An object containing a friendly message and a errorCode
+ */
+function getExtraFields (cartoError) {
+  var errorlist = _getErrorList(cartoError);
+  var listedError = _getListedError(cartoError, errorlist);
+
+  return {
+    friendlyMessage: listedError.friendlyMessage,
+    errorCode: listedError.errorCode
+  };
+}
+
+/**
+ * 
+ * @param {CartoError} cartoError 
+ */
+function _getErrorList (cartoError) {
+  return ERROR_LIST[cartoError.origin] && ERROR_LIST[cartoError.origin][cartoError.type];
+}
+
+/**
+ * Get the listed error from a cartoError, if no listedError is found return a generic
+ * unknown error.
+ * @param {CartoError} cartoError 
+ */
+function _getListedError (cartoError, errorList) {
+  var errorListkeys = _.keys(errorList);
+  var key;
+  for (var i = 0; i < errorListkeys.length; i++) {
+    key = errorListkeys[i];
+    if (errorList[key].messageRegex.test(cartoError.message)) {
+      return {
```

Comment:
I personally don't like exiting the function and the loop at the same time within the loop.

Reply:
Since we know each key is unique, we avoid iterating the half of the array.